crates/emitter/src/reference_op_emitter.rs

```diff
@@ -343,14 +343,14 @@ impl NameReferenceEmitter {
                 emitter.emit.get_g_name(name_index);
                 //      [stack] CALLEE
 
-                emitter.emit.g_implicit_this(name_index);
+                emitter.emit.undefined();
                 //      [stack] CALLEE THIS
             }
             NameLocation::Dynamic => {
                 emitter.emit.get_name(name_index);
                 //      [stack] CALLEE
 
-                emitter.emit.g_implicit_this(name_index);
+                emitter.emit.implicit_this(name_index);
                 //      [stack] CALLEE THIS
             }
```

Comment:
This does not seems to match what is implemented in https://phabricator.services.mozilla.com/D121278 ?

Reply:
that should've been `undefined`.
we omit other cases given we don't support non-syntactic scope or with statement.